samcli/commands/local/lib/swagger/parser.py

```diff
@@ -36,7 +60,187 @@ def get_binary_media_types(self):
+        auth_type: str
+            Type of authorizer (token, request)
+        event_type: str
+            API Gateway type (API, HTTP API)
+        properties: dict
+            Swagger Lambda Authorizer properties
+        authorizer_object: dict
+            Lambda Authorizer integration properties
+        Returns
+        -------
+        List[str]
+            A list of identity sources
+        """
+        identity_sources: List[str] = []
+
+        if auth_type == LambdaAuthorizer.TOKEN:
+            header_name = properties.get(SwaggerParser._AUTHORIZER_NAME)
+
+            if not properties.get(SwaggerParser._AUTHORIZER_IN) == "header" or not header_name:
+                LOG.info(
+                    "Missing properties for Lambda Authorizer '%s', "
+                    "property 'in' must be set to 'header' and "
+                    "property 'name' must be provided",
+                    auth_name,
+                )
+            elif event_type == Route.HTTP:
+                LOG.info("Type 'token' for Lambda Authorizer '%s' is unsupported ", auth_name)
+            else:
+                identity_sources.append(f"method.request.header.{header_name}")
+        else:
+            identity_source_string = authorizer_object.get(SwaggerParser._AUTHORIZER_IDENTITY_SOURCE)
+
+            if not identity_source_string:
+                LOG.info(
+                    "Missing property 'identitySource' in the authorizer integration for Lambda Authorizer '%s'",
+                    auth_name,
+                )
+            else:
+                # split the identity sources and remove any trailing spaces
+                split_identity_source: List[str] = identity_source_string.split(",")
+                identity_sources += [identity.strip() for identity in split_identity_source]
+
+        return identity_sources
+
+    def get_default_authorizer(self, event_type: str) -> Union[str, None]:
+        """
+        Parses the body definition to find root level Authorizer definitions
+
+        Parameters
+        ----------
+        event_type: str
+            String representing the type of API the definition body is defined as
+
+        Returns
+        -------
+        Union[str, None]
+            Returns the name of the authorizer, if there is one defined, otherwise None
+        """
+        document_version = self.swagger.get(SwaggerParser._SWAGGER) or self.swagger.get(SwaggerParser._OPENAPI) or ""
+        authorizers = self.swagger.get(SwaggerParser._SWAGGER_SECURITY, [])
+
+        if document_version.startswith(SwaggerParser._3_X_VERSION) and event_type == Route.HTTP:
+            if len(authorizers) > 1:
+                raise MultipleAuthorizerException(
+                    f"There must only be a single authorizer defined for a single route, found '{len(authorizers)}'"
+                )
+
+            if len(authorizers) == 1:
+                auth_name = str(list(authorizers[0])[0])
```

Comment:
discussed offline, we turn it into a list because the dictionary key is the auth name (so it's not a static key), but still we could do a check to ensure that list is not empty